src/util/ReactUtils.ts

```diff
@@ -97,11 +97,11 @@ export const toArray = <T extends ReactNode>(children: T | T[]): T[] => {
  * Find and return all matched children by type. `type` can be a React element class or
  * string
  */
-export const findAllByType = (
-  children: ReactNode,
-  type: string | string[],
-): React.DetailedReactHTMLElement<any, HTMLElement>[] => {
-  const result: React.DetailedReactHTMLElement<any, HTMLElement>[] = [];
+export function findAllByType<
+  Cmp extends React.ComponentType,
```

Comment:
Nit: Lets not use abbreviations. Instead, please write out the fields.
```suggestion
  Component extends React.ComponentType,
```